pymeasure/instruments/keithley/keithley2182.py

```diff
@@ -0,0 +1,390 @@
+        values={False: 0, True: 1},
+        map_values=True
+    )
+
+    def setup_voltage(self, auto_range=True, nplc=5):
+        """Set active channel and configure channel for voltage measurement.
+
+        :param auto_range: Enables auto_range if True, else uses set voltage
+            range
+        :param nplc: Number of power line cycles (NPLC) from 0.01 to 50/60
+        """
+        self.write(":SENS:CHAN {ch};"
+                   ":SENS:FUNC 'VOLT';"
+                   f":SENS:VOLT:NPLC {nplc};")
+        if auto_range:
+            self.write(":SENS:VOLT:RANG:AUTO 1")
+        self.check_errors()
+
+    def setup_temperature(self, nplc=5):
+        """Change active channel and configure channel for temperature measurement.
+
+        :param channel: Temperature measurement channel
+        :param nplc: Number of power line cycles (NPLC) from 0.01 to 50/60
+        """
+        self.write(":SENS:CHAN {ch};"
+                   ":SENS:FUNC 'TEMP';"
+                   f":SENS:TEMP:NPLC {nplc}")
+        self.check_errors()
+
+    def acquire_temperature_reference(self):
+        """Acquire a temperature measurement and store it as the relative offset value.
+
+        Only acquires reference if temperature offset is enabled.
+        """
+        self.write(":SENS:TEMP:CHAN{ch}:REF:ACQ")
+
+    def acquire_voltage_reference(self):
+        """Acquire a voltage measurement and store it as the relative offset value.
+
+        Only acquires reference if voltage offset is enabled.
+        """
+        self.write(":SENS:VOLT:CHAN{ch}:REF:ACQ")
+
+
+class Keithley2182(KeithleyBuffer, Instrument):
+    """Represents the Keithley 2182 Nanovoltmeter and provides a
+    high-level interface for interacting with the instrument.
+
+    .. code-block:: python
+
+        keithley = Keithley2182("GPIB::1")
+
+        keithley.reset()                        # Return instrument settings to default
+                                                  values
+        keithley.thermocouple = 'S'             # Sets thermocouple type to S
+        keithley.active_channel = 1             # Sets channel 1 for active measurement
+        keithley.channel_function = 'voltage'   # Configures active channel for voltage
+                                                  measurement
+        print(keithley.voltage)                 # Prints the voltage in volts
+
+        keithley.ch_1.setup_voltage()           # Set channel 1 active and prepare
+                                                  voltage measurement
+        keithley.ch_2.setup_temperature()       # Set channel 2 active and prepare
+                                                  temperature measurement
+
+    """
+
+    def __init__(self, adapter, name="Keithley 2182 Nanovoltmeter",
+                 read_termination='\r', **kwargs):
+        super().__init__(adapter, name, read_termination=read_termination,
+                         includeSCPI=True, **kwargs)
```

Comment:
Can you change the read_termination?
If not, you should not add it to the signature of the init, but hardcode it in the following way:
```suggestion
    def __init__(self, adapter, name="Keithley 2182 Nanovoltmeter",
                 **kwargs):
        super().__init__(adapter, name, read_termination="\r",
                         includeSCPI=True, **kwargs)
```

Reply:
The terminator can be CR, LF, or a combination of both. The default isn't listed in the instrument documentation, but ours is set to CR, so I've made that the default here.

Reply:
In this case, it is good as you did it.

A considerable amount of instruments have a fixed termination, where it useless, making it configurable. Therefore, I asked.